core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataReader.java

```diff
@@ -122,6 +128,17 @@ public Factory(FileSystemContext context, WorkerNetAddress address, long blockId
       mLocalReaderChunkSize = localReaderChunkSize;
       mReadBufferSize = conf.getInt(PropertyKey.USER_STREAMING_READER_BUFFER_SIZE_MESSAGES);
       mDataTimeoutMs = conf.getMs(PropertyKey.USER_STREAMING_DATA_TIMEOUT);
+      boolean isDirectMemoryIOEnabled = conf.getBoolean(PropertyKey.USER_DIRECT_MEMORY_IO_ENABLED);
+      if (isDirectMemoryIOEnabled) {
+        mBlockWorker = null;
+        mStream = null;
+        PropertyKey tierDirPathConf =
+            PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_PATH.format(0);
+        String ramdiskPath = conf.get(tierDirPathConf).split(",")[0];
+        String workerDir = conf.get(PropertyKey.WORKER_DATA_FOLDER);
+        mPath = Paths.get(ramdiskPath, workerDir, Long.toString(blockId)).toString();
```

Comment:
Is it required that this path is ramdisk? What happens with an SSD top-tier?

Reply:
toptier. name changed